Automattic/wp-calypso
# Simplify the canCurrentUser and isSiteUpgradeable selectors

Simplifies a few selectors that work with user capabilities.

First, I'm inlining the `isValidCapability` selector into `canCurrentUser` because it's used only there. The `isValidCapability` selector gets removed including its tests, because the `canCurrentUser` unit tests already test the same behavior.

The `canCurrentUser` now returns `null` when the `siteId` doesn't exist, instead of `false`. That's inline with how other selectors behave.

The `isSiteUpgradeable` selector doesn't need to call `getCurrentUser` or `getRawSite`. It can be just a wrapper around `canCurrentUser` and the return values don't change. If the user is logged out, the sites it gets from the anonymous `/sites` endpoints won't have the `capabilities` field, and `canCurrentUser` will return `null`. If the site doesn't exist, there won't be a `state.currentUser.capabilities[ siteId ]` record and again, the return value is `null`.

Finally, I'm removing the `is_customizable` and `isSiteUpgradeable` computed attributes on site objects. `is_customizable` is not used anywhere, and `isSiteUpgradeable` had just one usage that I replaced with `site.capabilities.manage_options` check.

**How to test:**
Verify that the changes are reasonable. Verify that the `site.isSiteUpgradeable` computed attribute is indeed never used. As it has the same name as the `isSiteUpgradeable` Redux selector, it's not easy to check.
